nav2_controller/src/nav2_controller.cpp

```diff
@@ -321,6 +322,13 @@ void ControllerServer::computeAndPublishVelocity()
     pose,
     nav_2d_utils::twist2Dto3D(twist));
 
+  std::shared_ptr<Action::Feedback> feedback = std::make_shared<Action::Feedback>();
+  feedback->speed = cmd_vel_2d.twist.linear.x;
+  feedback->distance_to_goal = sqrt(
```

Comment:
You could also use std::hypot here. It's slower, but I don't think that matters in this case and results in less code.